tempesta_fw/t/unit/test_tfw_str.c

```diff
@@ -314,101 +343,58 @@ TEST(tfw_str_eq_cstr, handles_empty_strs)
-	EXPECT_TRUE(tfw_str_eq_cstr(s, cstr2, len2, TFW_STR_EQ_CASEI));
-	EXPECT_TRUE(tfw_str_eq_cstr(s, cstr3, len3, TFW_STR_EQ_CASEI));
-	EXPECT_FALSE(tfw_str_eq_cstr(s, cstr2, len2, TFW_STR_EQ_DEFAULT));
-	EXPECT_FALSE(tfw_str_eq_cstr(s, cstr3, len3, TFW_STR_EQ_DEFAULT));
+	TFW_DBG("test_cassei:s:%s\n", s.data);
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, cstr1, len1, TFW_STR_EQ_CASEI));
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, cstr2, len2, TFW_STR_EQ_CASEI));
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, cstr3, len3, TFW_STR_EQ_CASEI));
+	EXPECT_FALSE(tfw_str_eq_cstr(&s, cstr2, len2, TFW_STR_EQ_DEFAULT));
+	EXPECT_FALSE(tfw_str_eq_cstr(&s, cstr3, len3, TFW_STR_EQ_DEFAULT));
 }
 
 TEST(tfw_str_eq_cstr, supports_prefix)
 {
-	TFW_STR(s, "/foo/bar/baz.test");
+	DEFINE_TFW_STR(s, "/foo/bar/baz.test");
 	const char *p1 = "/foo/bar/baz.test";
 	const char *p2 = "/foo/bar/baz.tes";
 	const char *p3 = "/foo/bar/baz";
 	const char *p4 = "/foo/bar/";
 	const char *p5 = "/foo";
 	const char *p6 = "/";
 	const char *p7 = "";
-	const char *extra = "/foo/bar/baz.test1";
-	const char *p1_ci = "/foo/bar/baz.tesT";
-	const char *p5_ci = "/Foo";
-
-	EXPECT_TRUE(tfw_str_eq_cstr(s, p1, strlen(p1), TFW_STR_EQ_PREFIX));
-	EXPECT_TRUE(tfw_str_eq_cstr(s, p2, strlen(p2), TFW_STR_EQ_PREFIX));
-	EXPECT_TRUE(tfw_str_eq_cstr(s, p3, strlen(p3), TFW_STR_EQ_PREFIX));
-	EXPECT_TRUE(tfw_str_eq_cstr(s, p4, strlen(p4), TFW_STR_EQ_PREFIX));
-	EXPECT_TRUE(tfw_str_eq_cstr(s, p5, strlen(p5), TFW_STR_EQ_PREFIX));
-	EXPECT_TRUE(tfw_str_eq_cstr(s, p6, strlen(p6), TFW_STR_EQ_PREFIX));
-	EXPECT_TRUE(tfw_str_eq_cstr(s, p7, strlen(p7), TFW_STR_EQ_PREFIX));
-
-	EXPECT_FALSE(tfw_str_eq_cstr(s, extra, strlen(extra),
+	const char *extra = "/foo/bar/baz.test/extra";
+	const char *p1_ci = "/foo/bar/";
+	const char *p5_ci = "/foo/bar/";
+	const char *p6_ci = "/foo/bar/p6_ci";
+
+	TFW_DBG("before tests:s:%s\n", s.data);
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, p1, strlen(p1), TFW_STR_EQ_PREFIX));
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, p2, strlen(p2), TFW_STR_EQ_PREFIX));
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, p3, strlen(p3), TFW_STR_EQ_PREFIX));
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, p4, strlen(p4), TFW_STR_EQ_PREFIX));
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, p5, strlen(p5), TFW_STR_EQ_PREFIX));
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, p6, strlen(p6), TFW_STR_EQ_PREFIX));
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, p7, strlen(p7), TFW_STR_EQ_PREFIX));
+
+	EXPECT_FALSE(tfw_str_eq_cstr(&s, extra, strlen(extra),
 		     TFW_STR_EQ_PREFIX));
-	EXPECT_FALSE(tfw_str_eq_cstr(s, p1_ci, strlen(p1_ci),
+	EXPECT_FALSE(tfw_str_eq_cstr(&s, p6_ci, strlen(p6_ci),
 		     TFW_STR_EQ_PREFIX));
-	EXPECT_FALSE(tfw_str_eq_cstr(s, p5_ci, strlen(p5_ci),
+	EXPECT_FALSE(tfw_str_eq_cstr(&s, p6_ci, strlen(p6_ci),
 		     TFW_STR_EQ_PREFIX));
 
-	EXPECT_TRUE(tfw_str_eq_cstr(s, p1_ci, strlen(p1_ci),
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, p1_ci, strlen(p1_ci),
 		    TFW_STR_EQ_PREFIX_CASEI));
-	EXPECT_TRUE(tfw_str_eq_cstr(s, p5_ci, strlen(p5_ci),
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, p5_ci, strlen(p5_ci),
```

Comment:
This is a strange piece of code. All these strings are constant. Please see the comment above regarding a similar piece of code.